interfaces/builtin/unity8.go

```diff
@@ -0,0 +1,161 @@
+# b) the list of url types is strictly controlled
+# c) the dispatched application will launch in the foreground over the
+#    confined app
+dbus (send)
+     bus=session
+     path=/com/canonical/URLDispatcher
+     interface=com.canonical.URLDispatcher
+     member=DispatchURL
+     peer=(name=com.canonical.URLDispatcher,label=###SLOT_SECURITY_TAGS###),
+
+# This is needed when the app is already running and needs to be passed in
+# a URL to open. This is most often used with content-hub providers and
+# url-dispatcher, but is actually supported by Qt generally.
+dbus (receive)
+     bus=session
+     path=/###PLUG_DBUS_APPIDS###
+     interface=org.freedesktop.Application
+     member=Open
+     peer=(label=###SLOT_SECURITY_TAGS###),
+
+# Unity launcher (e.g. app counter, progress, alert)
+dbus (receive, send)
+     bus=session
+     path=/com/canonical/Unity/Launcher/###PLUG_DBUS_APPIDS###
+     peer=(name=com.canonical.Unity.Launcher,label=###SLOT_SECURITY_TAGS###),
+
+# Note: content-hub may become its own interface, but for now include it here
+# Pasteboard via Content Hub. Unity8 with mir has safeguards that ensure snaps
+# only may get/set the pasteboard with user-driven actions.
+dbus (send)
+     bus=session
+     interface=com.ubuntu.content.dbus.Service
+     path=/
+     member={CreatePaste,GetAllPasteIds,GetLatestPasteData,GetPasteData,GetPasteSource,PasteFormats,RequestPasteByAppId,SelectPasteForAppId,SelectPasteForAppIdCancelled}
+     peer=(name=com.ubuntu.content.dbus.Service,label=###SLOT_SECURITY_TAGS###),
+dbus (receive)
+     bus=session
+     interface=com.ubuntu.content.dbus.Service
+     path=/
+     member={PasteboardChanged,PasteFormatsChanged,PasteSelected,PasteSelectionCancelled}
+     peer=(name=com.ubuntu.content.dbus.Service,label=###SLOT_SECURITY_TAGS###),
+
+# Lttng tracing is very noisy and should not be allowed by confined apps.
+# Can safely deny. LP: #1260491
+deny /{dev,run,var/run}/shm/lttng-ust-* r,
+`)
+
+var unity8ConnectedPlugSecComp = []byte(`
+shutdown
+`)
+
+type Unity8Interface struct{}
+
+func (iface *Unity8Interface) Name() string {
+	return "unity8"
+}
+
+func (iface *Unity8Interface) String() string {
+	return iface.Name()
+}
+
+func (iface *Unity8Interface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *Unity8Interface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		oldTags := []byte("###SLOT_SECURITY_TAGS###")
+		newTags := slotAppLabelExpr(slot)
+		snippet := bytes.Replace(unity8ConnectedPlugAppArmor, oldTags, newTags, -1)
+
+		// FIXME: Until we decide whether unity8 is going to use Snappy-style
+		//        appIDs (snap.NAME.COMMAND) or Touch-style ones
+		//        (NAME_COMMAND_REVISION), we'll use a simpler *NAME* glob
+		//        here.
```

Comment:
Please also add "This name glob does not provide isolation because 'foo' can access 'barfoobaz'. The base declaration restriction cannot be lifted until this is properly mediated."